.github/workflows/submit.yml

```diff
@@ -456,143 +461,14 @@ jobs:
-          --x-libraries=${HOME}/sysroot-arm64/usr/lib/aarch64-linux-gnu/
-          --with-build-jdk=${{ env.build_jdk_root }}
-          --with-conf-name=linux-aarch64
-          --with-version-opt=${GITHUB_ACTOR}-${GITHUB_SHA}
-          --with-version-build=0
-          --with-boot-jdk=${HOME}/bootjdk/${BOOT_JDK_VERSION}
-          --with-default-make-target="product-bundles test-bundles"
-          --with-zlib=system
-        working-directory: jdk
-
-      - name: Build
-        run: make CONF_NAME=linux-aarch64 ${{ matrix.build-target }}
-        working-directory: jdk
-
-  linux_arm_build:
-    name: Linux arm
-    runs-on: "ubuntu-20.04"
-    needs:
-      - prerequisites
-      - linux_x64_build
-    if: needs.prerequisites.outputs.should_run != 'false' && needs.prerequisites.outputs.platform_linux_arm != 'false'
-
-    strategy:
-      fail-fast: false
-      matrix:
-        flavor:
-          - build hotspot no-pch
-        include:
-          - flavor: build hotspot no-pch
-            flags: --enable-debug --disable-precompiled-headers
-            artifact: -debug
-            build-target: hotspot
-
-    env:
-      JDK_VERSION: "${{ fromJson(needs.prerequisites.outputs.dependencies).DEFAULT_VERSION_FEATURE }}"
-      BOOT_JDK_VERSION: "${{ fromJson(needs.prerequisites.outputs.dependencies).BOOT_JDK_VERSION }}"
-      BOOT_JDK_FILENAME: "${{ fromJson(needs.prerequisites.outputs.dependencies).LINUX_X64_BOOT_JDK_FILENAME }}"
-      BOOT_JDK_URL: "${{ fromJson(needs.prerequisites.outputs.dependencies).LINUX_X64_BOOT_JDK_URL }}"
-      BOOT_JDK_SHA256: "${{ fromJson(needs.prerequisites.outputs.dependencies).LINUX_X64_BOOT_JDK_SHA256 }}"
-
-    steps:
-      - name: Checkout the source
-        uses: actions/checkout@v2
-        with:
-          path: jdk
-
-      - name: Restore boot JDK from cache
-        id: bootjdk
-        uses: actions/cache@v2
-        with:
-          path: ~/bootjdk/${{ env.BOOT_JDK_VERSION }}
-          key: bootjdk-${{ runner.os }}-${{ env.BOOT_JDK_VERSION }}-${{ env.BOOT_JDK_SHA256 }}-v1
-
-      - name: Download boot JDK
-        run: |
-          mkdir -p "${HOME}/bootjdk/${BOOT_JDK_VERSION}"
-          wget -O "${HOME}/bootjdk/${BOOT_JDK_FILENAME}" "${BOOT_JDK_URL}"
-          echo "${BOOT_JDK_SHA256} ${HOME}/bootjdk/${BOOT_JDK_FILENAME}" | sha256sum -c >/dev/null -
-          tar -xf "${HOME}/bootjdk/${BOOT_JDK_FILENAME}" -C "${HOME}/bootjdk/${BOOT_JDK_VERSION}"
-          mv "${HOME}/bootjdk/${BOOT_JDK_VERSION}/"*/* "${HOME}/bootjdk/${BOOT_JDK_VERSION}/"
-        if: steps.bootjdk.outputs.cache-hit != 'true'
-
-      - name: Restore build JDK
-        id: build_restore
+      - name: Restore build JDK (retry)
         uses: actions/download-artifact@v2
         with:
           name: transient_jdk-linux-x64_${{ needs.prerequisites.outputs.bundle_id }}
           path: ~/jdk-linux-x64
-        continue-on-error: true
+        if: steps.build_restore.outcome == 'failure'
```

Comment:
I may be blind, but I don't see any step with id `build_restore` in new file...

Reply:
It already existed in the step right above so I didn't have to add it.. This follows the same pattern as other artifact downloads (I've seen them fail a fair amount of time in the past).

Reply:
Right. Got myself confused with GitHub UI.